bin/codemods/src/assign-then-export-react-class.js

```diff
@@ -0,0 +1,72 @@
+/*
+ This codemod updates
+
+ export default React.createClass( {
+    displayName: 'SomeDisplayName',
+ } );
+
+ to
+
+ const SomeDisplayName = React.createClass({
+   displayName: 'SomeDisplayName',
+ });
+
+ export default SomeDisplayName;
+ */
+
+/**
+ * External dependencies
+ */
+const { get } = require( 'lodash' );
+
+/**
+ * Module variables
+ */
+const createClassIdentifier = {
+    type: 'CallExpression',
+    callee: {
+        type: 'MemberExpression',
+        object: {
+            name: 'React',
+        },
+        property: {
+            name: 'createClass',
+        }
+    },
+};
+
+const displayNamePropertyIdentifier = {
+    key: {
+        name: 'displayName',
+    },
+};
+
+const DEFAULT_EXPORT_NAME = '__CHANGE_ME__';
+
+export default function transformer( file, api ) {
+    const j = api.jscodeshift;
+
+    const exportDefaultDeclaration = j( file.source )
+        .find( j.ExportDefaultDeclaration, {
+	         declaration: createClassIdentifier,
+        } );
+
+    const displayNamePaths = exportDefaultDeclaration.find( j.Property, displayNamePropertyIdentifier );
+    const displayNameValue = displayNamePaths.length && displayNamePaths.at( 0 ).get( 'value' );
+    const displayName = get( displayNameValue, 'value.value', DEFAULT_EXPORT_NAME );
```

Comment:
From @jblz: We could look at pulling out the `displayName` prop completely, since the displayName will now be inferred from the variable assignment.
Need to look at the current state of this - are there any risks?